src/main/java/org/radarcns/stream/collector/AggregateListCollector.java

```diff
@@ -16,26 +16,51 @@
 
 package org.radarcns.stream.collector;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.Arrays;
 import java.util.List;
+import org.apache.avro.Schema;
+import org.apache.avro.specific.SpecificRecord;
 
 /**
  * Java class to aggregate data using Kafka Streams. Double Array is the base type.
  */
-public class DoubleArrayCollector {
-    private DoubleValueCollector[] collectors;
+public class AggregateListCollector implements RecordCollector {
+    private final NumericAggregateCollector[] collectors;
+
+    @JsonCreator
+    public AggregateListCollector(
+            @JsonProperty("collectors") NumericAggregateCollector[] collectors) {
+        this.collectors = collectors;
+    }
+
+    /** Array collector without schema. Double entries can be added, but entire records cannot. */
+    public AggregateListCollector(String[] fieldNames) {
+        this(fieldNames, null);
+    }
+
+    /** Array collector with schema. Double entries or entire records can be added. */
+    public AggregateListCollector(String[] fieldNames, Schema schema) {
+        collectors = new NumericAggregateCollector[fieldNames.length];
+        for (int i = 0; i < collectors.length; i++) {
+            collectors[i] = new NumericAggregateCollector(fieldNames[i], schema);
```

Comment:
Do we assume that all the fields would have the same schema?

Reply:
Yes, I'll add that to the documentation.

Reply:
B.t.w., it's currently used with for example gyroscopes or accelerometers that have multiple axes.